internal/pkg/describe/env.go

```diff
@@ -50,23 +65,68 @@ func NewEnvDescriber(projectName string, envName string) (*EnvDescriber, error)
 		return nil, fmt.Errorf("assuming role for environment %s: %w", env.ManagerRoleARN, err)
 	}
 	return &EnvDescriber{
-		env:          env,
-		store:        svc,
-		apps:         apps,
-		sessProvider: sess,
+		env:      env,
+		store:    svc,
+		proj:     proj,
+		apps:     apps,
+		rgClient: resourcegroups.New(sess),
 	}, nil
 }
 
 // Describe returns info about a project's environment.
 func (e *EnvDescriber) Describe() (*EnvDescription, error) {
-	var tags map[string]string
+	appsForEnv, err := e.filterAppsForEnv()
+	if err != nil {
+		return nil, err
+	}
+
 	return &EnvDescription{
 		Environment:  e.env,
-		Applications: e.apps,
-		Tags:         tags,
+		Applications: appsForEnv,
+		Tags:         e.proj.Tags,
 	}, nil
 }
 
+func (e *EnvDescriber) filterAppsForEnv() ([]*archer.Application, error) {
+	var appObjects []*archer.Application
+
+	tags := map[string]string{
+		stack.EnvTagKey: e.env.Name,
+	}
+	arns, err := e.rgClient.GetResourcesByTags(cloudformationResourceType, tags)
+	if err != nil {
+		return nil, fmt.Errorf("get resources for env %s: %w", e.env.Name, err)
+	}
+
+	stacksOfEnvironment := make(map[string]bool)
+	for _, arn := range arns {
+		stack, err := e.getStackName(arn)
+		if err != nil {
+			return nil, fmt.Errorf("get stack name from ARN %s: %w", arn, err)
+		}
+		stacksOfEnvironment[stack] = true
+	}
+	for _, app := range e.apps {
+		stackName := stack.NameForApp(e.proj.Name, e.env.Name, app.Name)
+		if stacksOfEnvironment[stackName] {
+			appObjects = append(appObjects, app)
+		}
+	}
+	return appObjects, nil
+}
+
+func (e *EnvDescriber) getStackName(resourceArn string) (string, error) {
+	parsedArn, err := arn.Parse(resourceArn)
+	if err != nil {
+		return "", fmt.Errorf("parse ARN %s: %w", resourceArn, err)
+	}
+	stack := strings.Split(parsedArn.Resource, "/")
+	if len(stack) < 2 {
+		return "", fmt.Errorf("cannot parse ARN resource %s", parsedArn.Resource)
```

Comment:
nit: 
> When adding context to returned errors, keep the context succinct by avoiding phrases like "failed to", which state the obvious and pile up as the error percolates up through the stack

https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

We've been trying to avoid words like "cannot", "failed" in our error messages. Maybe:
```suggestion
		return "", fmt.Errorf("invalid ARN resource format %s", parsedArn.Resource)
```

Reply:
Can we also add a comment with what the ARN looks like so that a reader can understand why this would be invalid?